src/DynamoCore/Configuration/PathManager.cs

```diff
@@ -631,6 +632,44 @@ internal string GetUserDataFolder()
             return GetDynamoDataFolder(Path.Combine(folder, "Dynamo", "Dynamo Core"));
         }
 
+        /// <summary>
+        /// Returns the full path of user data location of all version of this
+        /// Dynamo product installed on this system. The default implementation
+        /// returns list of all subfolders in %appdata%\Dynamo as well as 
+        /// %appdata%\Dynamo\Dynamo Core\ folders.
+        /// </summary>
+        /// <returns></returns>
+        public virtual IEnumerable<string> GetDynamoUserDataLocations()
+        {
+            var appDatafolder = Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData);
+            var dynamoFolder = Path.Combine(appDatafolder, "Dynamo");
+            if (!Directory.Exists(dynamoFolder)) return Enumerable.Empty<string>();
+
+            var paths = new List<string>();
+            var coreFolder = Path.Combine(dynamoFolder, "Dynamo Core");
+            //Dynamo Core folder has to be enumerated first to cater migration from
+            //Dynamo 1.0 to Dynamo Core 1.0
+            if (Directory.Exists(coreFolder))
+            {
+                paths.AddRange(Directory.EnumerateDirectories(coreFolder));
+            }
+
+            paths.AddRange(Directory.EnumerateDirectories(dynamoFolder));
+            return paths;
+        }
+
+        /// <summary>
+        /// Returns the current Dynamo product version.
+        /// </summary>
+        /// <returns></returns>
+        public Updates.BinaryVersion GetProductVersion()
```

Comment:
This function is no longer static given PathManager is a Singleton now. The reason I did not move it to be a utility function is that we may call it multiple times in Dynamo life cycle and the old implementation is optimized to read the version only once.